src/protocol/sort/shuffle.rs

```diff
@@ -0,0 +1,99 @@
+use crate::helpers::prss::PrssSpace;
+
+#[allow(dead_code)]
+
+pub struct Shuffle {}
+
+impl Shuffle {
+    /// The Fisher–Yates shuffle is an algorithm for generating a random permutation of a finite
+    /// sequence—in plain terms, the algorithm shuffles the sequence. The algorithm effectively
+    /// puts all the elements into a hat; it continually determines the next element by randomly
+    /// drawing an element from the hat until no elements remain. The algorithm produces an unbiased
+    /// permutation: every permutation is equally likely. This algorithm takes time proportional to
+    /// the number of items being shuffled and shuffles them in place.
+    #[allow(dead_code)]
+    fn generate_random_sequence(
+        batchsize: usize,
+        random_value: u128,
+        left_permute: bool,
+        prss: &PrssSpace,
+    ) -> Vec<usize> {
+        let mut permutation: Vec<usize> = (0..batchsize).collect();
+        (1..batchsize).rev().for_each(|i| {
+            let location = if left_permute {
+                (prss.generate_values(random_value).0 as usize) % i
```

Comment:
there is no reason to call this in a loop, it returns the same value over and over. Moreover this call is compute-heavy as it uses a block cipher to encrypt the index. 

But the bigger question is whether it is secure. This proposal uses a single pseudo-random value to derive the whole permutation, so we are not drawing from a "hat" n times

Reply:
Replacing it completely with a rust library function.